pkg/sdn/plugin/subnets.go

```diff
@@ -155,6 +165,37 @@ func (master *OsdnMaster) watchNodes() {
 	}
 }
 
+// Because openshift-sdn uses an overlay and doesn't need GCE Routes,
+// we need to clear the kubelet-added NetworkUnavailable condition
+// that kubelet adds to initial node status when using GCE.
+// TODO: make upstream kubelet more flexible with overlays and GCE
+func (node *OsdnNode) clearInitialNodeNetworkingCondition() error {
+	knode, err := node.registry.GetNode(node.hostName)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < updateNodeStatusMaxRetries; i++ {
+		_, condition := kapi.GetNodeCondition(&knode.Status, kapi.NodeNetworkUnavailable)
+		if condition != nil && condition.Status != kapi.ConditionFalse && condition.Reason == "NoRouteCreated" {
+			condition.Status = kapi.ConditionFalse
+			condition.Reason = "RouteCreated"
+			condition.Message = "openshift-sdn cleared kubelet-set NoRouteCreated"
+			condition.LastTransitionTime = kapiunversioned.Now()
+		}
+
+		if _, err = node.registry.UpdateNodeStatus(knode); err == nil {
```

Comment:
can we be more judicious on retrying here? are we retrying for conflict errors? notfound errors?